apis/low_level_debug/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+#![no_std]
+
+use libtock_platform::{CommandReturn, Syscalls};
+
+/// The low-level debug API provides tools to diagnose userspace issues that make normal debugging workflows (e.g. printing to the console) difficult.
+///
+/// It allows libraries to print alert codes and apps to print numeric information using only the command system call.
+///
+/// # Example
+/// ```ignore
+/// use libtock_runtime::TockSyscalls;
+///
+/// // Uses the real syscall implementation.
+/// type LowLevelDebug = libtock_low_level_debug::LowLevelDebug<TockSyscalls>;
+///
+/// // Prints 0x45 and the app which called it.
+/// LowLevelDebug::print_1(0x45);
+/// ```
+pub struct LowLevelDebug<S: Syscalls>(S);
+
+impl<S: Syscalls> LowLevelDebug<S> {
+    /// Run a check against the low-level debug capsule to ensure it is working.
+    #[inline(always)]
+    pub fn driver_check() -> CommandReturn {
+        S::command(DRIVER_ID, DRIVER_CHECK, 0, 0)
+    }
+
+    /// Print one of the predefined alerts in [`AlertCode`].
+    #[inline(always)]
+    pub fn print_alert_code(code: AlertCode) -> CommandReturn {
+        S::command(DRIVER_ID, PRINT_ALERT_CODE, code as u32, 0)
+    }
+
+    /// Print a single number. The number will be printed in hexadecimal.
+    ///
+    /// In general, this should only be added temporarily for debugging and should not be called by released library code.
+    #[inline(always)]
+    pub fn print_1(x: u32) -> CommandReturn {
+        S::command(DRIVER_ID, PRINT_1, x, 0)
+    }
+
+    /// Print two numbers. The numbers will be printed in hexadecimal.
+    ///
+    /// Like `print_1`, this is intended for temporary debugging and should not be called by released library code.
+    /// If you want to print multiple values, it is often useful to use the first argument to indicate what value is being printed.
+    #[inline(always)]
+    pub fn print_2(x: u32, y: u32) -> CommandReturn {
+        S::command(DRIVER_ID, PRINT_2, x, y)
+    }
+}
+
+/// A predefined alert code, for use with [`LowLevelDebug::print_alert_code`].
+///
+/// Predefined alert codes are intended for use in library code, and are defined here to avoid collisions between projects.
+#[repr(u32)]
```

Comment:
Is `#[repr(u32)]` benificial here? It seems unnecessary to me.

Reply:
Not particularly - I initially put it to make it clear that they were used as u32's, but that's an unimportant implementation detail.